go_modules/spec/dependabot/go_modules/file_parser_spec.rb

```diff
@@ -10,6 +10,13 @@
 RSpec.describe Dependabot::GoModules::FileParser do
   it_behaves_like "a dependency file parser"
 
+  before(:all) do
+    ENV["LEGACY_GO"] = ENV.fetch("GOTOOLCHAIN")
+  end
+  after(:all) do
+    ENV["GOTOOLCHAIN"] = ENV.fetch("LEGACY_GO")
+  end
```

Comment:
The above hooks only run once before and after running all tests in this file. I generally tend to avoid it, except for very expensive setups. This seems only related to the test being added, right? Can we instead move it there?

Reply:
I was seeing a flake where a test in another suite was failing because the GOTOOLCHAIN was still set from what seemed to be a test in this suite. Adding this fixed it, maybe I should remove `before(:all)` and `after(:all)` and just do an `after` here so I can ensure the order of any of these doesn't affect the other tests inside or outside the suite?

Reply:
> I was seeing a flake where a test in another suite was failing because the GOTOOLCHAIN was still set from what seemed to be a test in this suite. Adding this fixed it, maybe I should remove after(:all) and just do an after here so I can ensure the order of any of these doesn't affect the other tests inside or outside the suite?

Yes, I think ideally each test that makes a change over the default `GOTOOLCHAIN` should clean up after itself and not leave any side effects. So I think removing the `:all` and doing this for every test would work?

Reply:
Sorry our messages crossed. I'll review the new changes.

Reply:
I think it's good to review now!